### PR TITLE
Fixed and enabled some of the tcp tests

### PR DESCRIPTION
**Pull request by stevenengler**
Related: shadow/shadow#974

### CODE REVIEW

**Comment by florian-vuillemot** [comment]
Would you like me to give you a hand in changing the tests in the "new way"? I got used to deal with it ^^

**Comment by sporksmith** [comment]
Oops, sorry I missed that a review was requested here. I wish github made it easier to see a list of all pending requested reviews, rather than just notifications on state-changes.

Should I wait for conflict resolution?

**Comment by stevenengler** [comment]
> Oops, sorry I missed that a review was requested here. I wish github made it easier to see a list of all pending requested reviews, rather than just notifications on state-changes.
>
> Should I wait for conflict resolution?

@sporksmith Sure, I wasn't sure if you had started looking at it yet, so it sounds easiest for me to just fix it now.

> Would you like me to give you a hand in changing the tests in the "new way"? I got used to deal with it ^^

@florian-vuillemot Thanks for offering to help, I think it would be quickest for me to update these since I'm going to squash it into one of the previous commits.

**Comment by robgjansen** [comment]
> Oops, sorry I missed that a review was requested here. I wish github made it easier to see a list of all pending requested reviews, rather than just notifications on state-changes.

@sporksmith This?
https://github.com/pulls/review-requested

**Comment by sporksmith** [comment]
> > Oops, sorry I missed that a review was requested here. I wish github made it easier to see a list of all pending requested reviews, rather than just notifications on state-changes.
> 
> @sporksmith This?
> https://github.com/pulls/review-requested

Ah ha! Thanks :)

File: src/test/tcp/CMakeLists.txt [hunk 1]
```diff
@@ -3,151 +3,29 @@ link_libraries(${GLIB_LIBRARIES})
...
-    | ${CMAKE_SOURCE_DIR}/src/test/tcp/with_q.sh ${CMAKE_BINARY_DIR}/src/main/shadow --interpose-method=ptrace -l debug -d iov.shadow-ptrace.data - \
-    "
-)
-# FIXME add preload test
-
-set_tests_properties(
-  tcp-blocking-loopback tcp-nonblocking-poll-loopback tcp-nonblocking-epoll-loopback tcp-nonblocking-select-loopback tcp-iov
-  PROPERTIES RUN_SERIAL true
-)
+# FIXME: enable other blocking modes
```
Comment (sporksmith) [inline src/test/tcp/CMakeLists.txt:6]: Much easier to follow with this structure :)

**Review by sporksmith — APPROVED** [review]

File: src/test/tcp/test_tcp.c [hunk 2]
```diff
@@ -710,9 +682,33 @@ static int _test_iov_server(int clientfd)
...
+    *error = 0;
+
+    char* endptr;
+    errno = 0;
+    long server_port_long = strtol(str, &endptr, 10);
+    // Check for various possible errors
+    if ((errno == ERANGE && (server_port_long == LONG_MAX || server_port_long == LONG_MIN)) ||
+        (errno != 0 && server_port_long == 0) || endptr == str) {
+        MYLOG("error, could not parse port number");
+        perror("strtol");
```
[the remark this answers is not in the document]
Comment (stevenengler) [reply src/test/tcp/test_tcp.c]: Fixed, thanks!

**Pull request merged**